File: SourceCppProject/sources/Parsers/GeneralCharacterValidatorTest.cpp
#include <gtest/gtest.h>

#include "GeneralCharacterValidator.hpp"


TEST(GeneralCharacterValidatorTest, IsNotCommentCharacter_TestCase_1)
{
    const char comment_character {'#'};
    const char test_character {'A'};

    EXPECT_EQ(GeneralCharacterValidator::isNotCommentCharacter(test_character), test_character != comment_character);
}

TEST(GeneralCharacterValidatorTest, IsNotCommentCharacter_TestCase_2)
{
    const char comment_character {'#'};
    const char test_character {'Z'};

    EXPECT_EQ(GeneralCharacterValidator::isNotCommentCharacter(test_character), test_character != comment_character);
}

TEST(GeneralCharacterValidatorTest, IsNotCommentCharacter_TestCase_3)
{
    const char comment_character {'#'};
    const char test_character {'b'};

    EXPECT_EQ(GeneralCharacterValidator::isNotCommentCharacter(test_character), test_character != comment_character);
}

TEST(GeneralCharacterValidatorTest, IsNotCommentCharacter_TestCase_4)
{
    const char comment_character {'#'};
    const char test_character {'y'};

    EXPECT_EQ(GeneralCharacterValidator::isNotCommentCharacter(test_character), test_character != comment_character);
}

TEST(GeneralCharacterValidatorTest, IsNotCommentCharacter_TestCase_5)
{
    const char comment_character {'#'};
    const char test_character {'1'};

    EXPECT_EQ(GeneralCharacterValidator::isNotCommentCharacter(test_character), test_character != comment_character);
}

TEST(GeneralCharacterValidatorTest, IsNotCommentCharacter_TestCase_6)
{
    const char comment_character {'#'};
    const char test_character {'9'};

    EXPECT_EQ(GeneralCharacterValidator::isNotCommentCharacter(test_character), test_character != comment_character);
}

TEST(GeneralCharacterValidatorTest, IsNotCommentCharacter_TestCase_7)
{
    const char comment_character {'#'};
    const char test_character {'@'};

    EXPECT_EQ(GeneralCharacterValidator::isNotCommentCharacter(test_character), test_character != comment_character);
}

TEST(GeneralCharacterValidatorTest, IsNotCommentCharacter_TestCase_8)
{
    const char comment_character {'#'};
    const char test_character {'&'};

    EXPECT_EQ(GeneralCharacterValidator::isNotCommentCharacter(test_character), test_character != comment_character);
}


TEST(GeneralCharacterValidatorTest, IsCommentCharacter)
{
    const char comment_character {'#'};
    const char test_character {comment_character};

    EXPECT_EQ(GeneralCharacterValidator::isNotCommentCharacter(test_character), test_character != comment_character);
}


TEST(GeneralCharacterValidatorTest, IsSpaceCharacter_TestCase_1)
{
    EXPECT_EQ(GeneralCharacterValidator::isSpaceCharacter(' '), true); // ASCII: 0x20
}

TEST(GeneralCharacterValidatorTest, IsSpaceCharacter_TestCase_2)
{
    EXPECT_EQ(GeneralCharacterValidator::isSpaceCharacter('\f'), true); // ASCII: 0x0C
}

TEST(GeneralCharacterValidatorTest, IsSpaceCharacter_TestCase_3)
{
    EXPECT_EQ(GeneralCharacterValidator::isSpaceCharacter('\n'), true); // ASCII: 0x0A
}

TEST(GeneralCharacterValidatorTest, IsSpaceCharacter_TestCase_4)
{
    EXPECT_EQ(GeneralCharacterValidator::isSpaceCharacter('\r'), true); // ASCII: 0x0D
}

TEST(GeneralCharacterValidatorTest, IsSpaceCharacter_TestCase_5)
{
    EXPECT_EQ(GeneralCharacterValidator::isSpaceCharacter('\t'), true); // ASCII: 0x09
}

TEST(GeneralCharacterValidatorTest, IsSpaceCharacter_TestCase_6)
{
    EXPECT_EQ(GeneralCharacterValidator::isSpaceCharacter('\v'), true); // ASCII: 0x0B
}


TEST(GeneralCharacterValidatorTest, IsNotSpaceCharacter_TestCase_1)
{
    EXPECT_EQ(GeneralCharacterValidator::isSpaceCharacter('A'), false);
}

TEST(GeneralCharacterValidatorTest, IsNotSpaceCharacter_TestCase_2)
{
    EXPECT_EQ(GeneralCharacterValidator::isSpaceCharacter('Z'), false);
}

TEST(GeneralCharacterValidatorTest, IsNotSpaceCharacter_TestCase_3)
{
    EXPECT_EQ(GeneralCharacterValidator::isSpaceCharacter('b'), false);
}

TEST(GeneralCharacterValidatorTest, IsNotSpaceCharacter_TestCase_4)
{
    EXPECT_EQ(GeneralCharacterValidator::isSpaceCharacter('y'), false);
}

TEST(GeneralCharacterValidatorTest, IsNotSpaceCharacter_TestCase_5)
{
    EXPECT_EQ(GeneralCharacterValidator::isSpaceCharacter('0'), false);
}

TEST(GeneralCharacterValidatorTest, IsNotSpaceCharacter_TestCase_6)
{
    EXPECT_EQ(GeneralCharacterValidator::isSpaceCharacter('9'), false);
}

TEST(GeneralCharacterValidatorTest, IsNotSpaceCharacter_TestCase_7)
{
    EXPECT_EQ(GeneralCharacterValidator::isSpaceCharacter('#'), false);
}

TEST(GeneralCharacterValidatorTest, IsNotSpaceCharacter_TestCase_8)
{
    EXPECT_EQ(GeneralCharacterValidator::isSpaceCharacter('/'), false);
}

TEST(GeneralCharacterValidatorTest, IsNotSpaceCharacter_TestCase_9)
{
    EXPECT_EQ(GeneralCharacterValidator::isSpaceCharacter('\\'), false);
}

File: TextLinesComparing.Testing/GeneralCharacterValidatorTest.cs
using TextLinesComparing.Library;

namespace TextLinesComparing.Testing;

public class GeneralCharacterValidatorTest
{
    [Test]
    public void IsCommentCharacter()
    {
        const char comment_character = '#';
        const char test_character = comment_character;

        Assert.That(GeneralCharacterValidator.IsNotCommentCharacter(test_character), Is.EqualTo(test_character != comment_character));
    }

    [Test]
    public void IsNotCommentCharacter_TestCase()
    {
        char comment_character = '#';
        char test_character = default;
        Assert.That(GeneralCharacterValidator.IsNotCommentCharacter(test_character), Is.EqualTo(test_character != comment_character));

        comment_character = '#';
        test_character = 'Z';
        Assert.That(GeneralCharacterValidator.IsNotCommentCharacter(test_character), Is.EqualTo(test_character != comment_character));

        comment_character = '#';
        test_character = 'b';
        Assert.That(GeneralCharacterValidator.IsNotCommentCharacter(test_character), Is.EqualTo(test_character != comment_character));

        comment_character = '#';
        test_character = 'y';
        Assert.That(GeneralCharacterValidator.IsNotCommentCharacter(test_character), Is.EqualTo(test_character != comment_character));

        comment_character = '#';
        test_character = '1';
        Assert.That(GeneralCharacterValidator.IsNotCommentCharacter(test_character), Is.EqualTo(test_character != comment_character));

        comment_character = '#';
        test_character = '9';
        Assert.That(GeneralCharacterValidator.IsNotCommentCharacter(test_character), Is.EqualTo(test_character != comment_character));

        comment_character = '#';
        test_character = '@';
        Assert.That(GeneralCharacterValidator.IsNotCommentCharacter(test_character), Is.EqualTo(test_character != comment_character));

        comment_character = '#';
        test_character = '&';
        Assert.That(GeneralCharacterValidator.IsNotCommentCharacter(test_character), Is.EqualTo(test_character != comment_character));
    }

    [Test]
    public void IsSpaceCharacter_TestCase()
    {
        Assert.Multiple(() =>
        {
            Assert.That(GeneralCharacterValidator.IsSpaceCharacter(' '), Is.True); // ASCII: 0x20
            Assert.That(GeneralCharacterValidator.IsSpaceCharacter('\f'), Is.True); // ASCII: 0x0C
            Assert.That(GeneralCharacterValidator.IsSpaceCharacter('\n'), Is.True); // ASCII: 0x0A
            Assert.That(GeneralCharacterValidator.IsSpaceCharacter('\r'), Is.True); // ASCII: 0x0D
            Assert.That(GeneralCharacterValidator.IsSpaceCharacter('\t'), Is.True); // ASCII: 0x09
            Assert.That(GeneralCharacterValidator.IsSpaceCharacter('\v'), Is.True); // ASCII: 0x0B
            Assert.That(GeneralCharacterValidator.IsSpaceCharacter('A'), Is.False);
            Assert.That(GeneralCharacterValidator.IsSpaceCharacter('Z'), Is.False);
            Assert.That(GeneralCharacterValidator.IsSpaceCharacter('b'), Is.False);
            Assert.That(GeneralCharacterValidator.IsSpaceCharacter('y'), Is.False);
            Assert.That(GeneralCharacterValidator.IsSpaceCharacter('0'), Is.False);
            Assert.That(GeneralCharacterValidator.IsSpaceCharacter('9'), Is.False);
            Assert.That(GeneralCharacterValidator.IsSpaceCharacter('#'), Is.False);
            Assert.That(GeneralCharacterValidator.IsSpaceCharacter('/'), Is.False);
            Assert.That(GeneralCharacterValidator.IsSpaceCharacter('\\'), Is.False);
        });
    }
}
